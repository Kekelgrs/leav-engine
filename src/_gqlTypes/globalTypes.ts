/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum ActionIOTypes {
    boolean = 'boolean',
    number = 'number',
    object = 'object',
    string = 'string'
}

export enum AttributeFormat {
    boolean = 'boolean',
    date = 'date',
    encrypted = 'encrypted',
    extended = 'extended',
    numeric = 'numeric',
    text = 'text'
}

export enum AttributeType {
    advanced = 'advanced',
    advanced_link = 'advanced_link',
    simple = 'simple',
    simple_link = 'simple_link',
    tree = 'tree'
}

export enum AvailableActionsName {
    encrypt = 'encrypt',
    formatDate = 'formatDate',
    formatNumber = 'formatNumber',
    maskValue = 'maskValue',
    parseJSON = 'parseJSON',
    toBoolean = 'toBoolean',
    toJSON = 'toJSON',
    toNumber = 'toNumber',
    toString = 'toString',
    validateFormat = 'validateFormat',
    validateRegex = 'validateRegex'
}

export enum AvailableLanguage {
    en = 'en',
    fr = 'fr'
}

export enum PermissionTypes {
    admin = 'admin',
    attribute = 'attribute',
    library = 'library',
    record = 'record'
}

export enum PermissionsActions {
    access = 'access',
    access_attribute = 'access_attribute',
    admin_access_attributes = 'admin_access_attributes',
    admin_access_libraries = 'admin_access_libraries',
    admin_access_navigator = 'admin_access_navigator',
    admin_access_permissions = 'admin_access_permissions',
    admin_access_trees = 'admin_access_trees',
    admin_create_attribute = 'admin_create_attribute',
    admin_create_library = 'admin_create_library',
    admin_create_tree = 'admin_create_tree',
    admin_delete_attribute = 'admin_delete_attribute',
    admin_delete_library = 'admin_delete_library',
    admin_delete_tree = 'admin_delete_tree',
    admin_edit_attribute = 'admin_edit_attribute',
    admin_edit_library = 'admin_edit_library',
    admin_edit_permission = 'admin_edit_permission',
    admin_edit_tree = 'admin_edit_tree',
    create = 'create',
    create_value = 'create_value',
    delete = 'delete',
    delete_value = 'delete_value',
    edit = 'edit',
    edit_value = 'edit_value'
}

export enum PermissionsRelation {
    and = 'and',
    or = 'or'
}

export enum ValueVersionMode {
    simple = 'simple',
    smart = 'smart'
}

export interface ActionConfigurationInput {
    name: AvailableActionsName;
    params?: (ActionConfigurationParamInput | null)[] | null;
}

export interface ActionConfigurationParamInput {
    name: string;
    value: string;
}

export interface ActionsListConfigurationInput {
    saveValue?: (ActionConfigurationInput | null)[] | null;
    getValue?: (ActionConfigurationInput | null)[] | null;
    deleteValue?: (ActionConfigurationInput | null)[] | null;
}

export interface AttributeInput {
    id: string;
    type?: AttributeType | null;
    format?: AttributeFormat | null;
    label?: SystemTranslationInput | null;
    linked_library?: string | null;
    linked_tree?: string | null;
    embedded_fields?: (EmbeddedAttributeInput | null)[] | null;
    actions_list?: ActionsListConfigurationInput | null;
    permissions_conf?: Treepermissions_confInput | null;
    multiple_values?: boolean | null;
    versions_conf?: ValuesVersionsConfInput | null;
}

export interface EmbeddedAttributeInput {
    id: string;
    format?: AttributeFormat | null;
    label?: SystemTranslationInput | null;
    validation_regex?: string | null;
    embedded_fields?: (EmbeddedAttributeInput | null)[] | null;
}

export interface LibraryInput {
    id: string;
    label?: SystemTranslationInput | null;
    attributes?: string[] | null;
    permissions_conf?: Treepermissions_confInput | null;
    recordIdentityConf?: RecordIdentityConfInput | null;
}

export interface PermissionActionInput {
    name: PermissionsActions;
    allowed?: boolean | null;
}

/**
 * If users group is not specified, permission will be saved at root level.
 * If saving a tree-based permission (record or attribute) and tree target's id is not specified,
 * permission will be saved at root level for any element of the tree.
 */
export interface PermissionInput {
    type: PermissionTypes;
    applyTo?: string | null;
    usersGroup?: string | null;
    actions: PermissionActionInput[];
    permissionTreeTarget?: PermissionsTreeTargetInput | null;
}

/**
 * Element on which we want to retrieve record or attribute permission. Record ID is mandatory,
 * attributeId is only required for attribute permission
 */
export interface PermissionTarget {
    attributeId?: string | null;
    recordId: string;
}

/**
 * If id and library are not specified, permission will apply to tree root
 */
export interface PermissionsTreeTargetInput {
    tree: string;
    library?: string | null;
    id?: string | null;
}

export interface RecordIdentityConfInput {
    label?: string | null;
    color?: string | null;
    preview?: string | null;
}

export interface SystemTranslationInput {
    fr: string;
    en?: string | null;
}

export interface TreeElementInput {
    id: string;
    library: string;
}

export interface TreeInput {
    id: string;
    libraries?: string[] | null;
    label?: SystemTranslationInput | null;
}

export interface Treepermissions_confInput {
    permissionTreeAttributes: string[];
    relation: PermissionsRelation;
}

export interface ValueBatchInput {
    attribute?: string | null;
    id_value?: string | null;
    value?: string | null;
}

export interface ValueVersionInput {
    name: string;
    value: TreeElementInput;
}

export interface ValuesVersionsConfInput {
    versionable: boolean;
    mode?: ValueVersionMode | null;
    trees?: string[] | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
