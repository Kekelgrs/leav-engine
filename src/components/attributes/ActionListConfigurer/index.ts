import ActionListConfigurer from './ActionListConfigurer';
export default ActionListConfigurer;
