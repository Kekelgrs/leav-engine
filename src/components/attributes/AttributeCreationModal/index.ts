import AttributeCreationModal from './AttributeCreationModal';
export default AttributeCreationModal;
