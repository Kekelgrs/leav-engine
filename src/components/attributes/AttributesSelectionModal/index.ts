import AttributesSelectionModal from './AttributesSelectionModal';
export default AttributesSelectionModal;
