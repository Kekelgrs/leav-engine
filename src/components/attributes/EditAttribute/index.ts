import EditAttribute from './EditAttribute';
export default EditAttribute;
