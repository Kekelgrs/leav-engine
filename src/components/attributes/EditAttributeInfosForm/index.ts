import EditAttributeInfosForm from './EditAttributeInfosForm';
export default EditAttributeInfosForm;
