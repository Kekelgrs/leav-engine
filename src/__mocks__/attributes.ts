import {GET_ATTRIBUTES_attributes_list} from '../_gqlTypes/GET_ATTRIBUTES';
import {AttributeFormat, AttributeType, ValueVersionMode} from '../_gqlTypes/globalTypes';

const base: GET_ATTRIBUTES_attributes_list = {
    id: 'test_attribute',
    label: {
        fr: 'Mon Attribut',
        en: 'My Attribute'
    },
    type: AttributeType.simple,
    format: AttributeFormat.text,
    multiple_values: false,
    system: false,
    linked_tree: null,
    linked_library: null,
    permissions_conf: null,
    versions_conf: {versionable: false, mode: ValueVersionMode.smart, trees: null},
    metadata_fields: null
};

export const mockAttrSimple = {...base, id: 'simple_attribute'};
export const mockAttrId = {...base, id: 'id', system: true};
export const mockAttrSimpleWithValuesList = {
    ...mockAttrSimple,
    id: 'simple_attribute_with_values_list',
    values_list: {
        enable: true,
        allowFreeEntry: false,
        values: ['value 1', 'value 2']
    }
};

export const mockAttrAdv = {...base, id: 'advanced_attribute', type: AttributeType.advanced};
export const mockAttrAdvMultiVal = {...mockAttrAdv, multiple_values: true};
export const mockAttrAdvVersionable = {
    ...mockAttrAdv,
    versions_conf: {versionable: true, trees: ['tree_attribute']}
};

export const mockAttrSimpleLink = {...base, id: 'simple_link_attribute', type: AttributeType.simple_link};
export const mockAttrSimpleLinkWithValuesList = {
    ...mockAttrSimpleLink,
    id: 'simple_link_attribute_with_values_list',
    values_list: {
        enable: true,
        allowFreeEntry: false,
        values: ['132456', '987654']
    }
};

export const mockAttrAdvLink = {...base, id: 'adv_link_attribute', type: AttributeType.advanced_link};
export const mockAttrAdvLinkMultiVal = {...mockAttrAdvLink, multiple_values: true};
export const mockAttrAdvLinkWithValuesList = {
    ...mockAttrAdvLink,
    id: 'adv_link_attribute_with_values_list',
    values_list: {
        enable: true,
        allowFreeEntry: false,
        values: ['132456', '987654']
    }
};

export const mockAttrTree = {...base, id: 'tree_attribute', type: AttributeType.tree};
export const mockAttrTreeMultival = {...mockAttrTree, multiple_values: true};
