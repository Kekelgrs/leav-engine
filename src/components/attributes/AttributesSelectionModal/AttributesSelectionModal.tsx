import React from 'react';
import {withNamespaces, WithNamespaces} from 'react-i18next';
import {Button, Modal} from 'semantic-ui-react';
import {AttributesQuery, getAttributesQuery} from '../../../queries/attributes/getAttributesQuery';
import {GET_ATTRIBUTES_attributes_list} from '../../../_gqlTypes/GET_ATTRIBUTES';
import Loading from '../../shared/Loading';
import AttributesSelectionList from '../AttributesSelectionList';

interface IAttributesSelectionProps extends WithNamespaces {
    onSubmit: (selectedAttributes: string[]) => void;
    openModal: boolean;
    onClose: () => void;
    selection: string[];
}

interface IAttributesSelectionState {
    pendingSelection: string[];
}

class AttributesSelectionModal extends React.Component<IAttributesSelectionProps, IAttributesSelectionState> {
    constructor(props: IAttributesSelectionProps) {
        super(props);

        this.state = {
            pendingSelection: []
        };
    }

    public render() {
        const {t, openModal, selection} = this.props;
        const {pendingSelection} = this.state;
        return (
            openModal && (
                <AttributesQuery query={getAttributesQuery}>
                    {({loading, error, data}) => {
                        if (loading || !data) {
                            return <Loading />;
                        }
                        if (typeof error !== 'undefined') {
                            return <p>Error: {error.message}</p>;
                        }

                        return (
                            data.attributes && (
                                <Modal
                                    size="small"
                                    open={openModal}
                                    onClose={this._handleclose}
                                    centered
                                    closeOnDimmerClick
                                    closeOnEscape
                                >
                                    <Modal.Header>{t('libraries.link_existing_attribute')}</Modal.Header>
                                    <Modal.Content scrolling>
                                        <AttributesSelectionList
                                            attributes={data.attributes.list.filter(
                                                a => selection.indexOf(a.id) === -1
                                            )}
                                            selection={pendingSelection}
                                            toggleSelection={this._toggleSelection}
                                        />
                                    </Modal.Content>
                                    <Modal.Actions>
                                        <Button onClick={this._handleclose}>{t('admin.cancel')}</Button>
                                        <Button onClick={this._handleSubmit}>{t('admin.submit')}</Button>
                                    </Modal.Actions>
                                </Modal>
                            )
                        );
                    }}
                </AttributesQuery>
            )
        );
    }

    private _handleclose = () => {
        this.setState({pendingSelection: []});
        this.props.onClose();
    }

    private _handleSubmit = () => {
        this.props.onSubmit(this.state.pendingSelection);
    }

    private _toggleSelection = (selectedAttr: GET_ATTRIBUTES_attributes_list) => {
        const {pendingSelection} = this.state;

        const newSelection =
            pendingSelection.indexOf(selectedAttr.id) !== -1
                ? pendingSelection.filter(aId => aId !== selectedAttr.id)
                : [...pendingSelection, selectedAttr.id];

        this.setState({pendingSelection: newSelection});
    }
}

export default withNamespaces()(AttributesSelectionModal);
