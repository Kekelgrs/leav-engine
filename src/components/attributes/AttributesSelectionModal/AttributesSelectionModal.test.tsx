import {MockedProvider, wait} from '@apollo/react-testing';
import {mount} from 'enzyme';
import React from 'react';
import {act} from 'react-dom/test-utils';
import {getAttributesQuery} from '../../../queries/attributes/getAttributesQuery';
import {AttributeType} from '../../../_gqlTypes/globalTypes';
import {mockAttrSimple} from '../../../__mocks__/attributes';
import AttributesSelectionModal from './AttributesSelectionModal';

describe('AttributesSelection', () => {
    test('Snapshot test', async () => {
        const mocks = [
            {
                request: {
                    query: getAttributesQuery,
                    variables: {type: [AttributeType.tree]}
                },
                result: {
                    data: {
                        attributes: {
                            __typename: 'AttributesList',
                            totalCount: 0,
                            list: [
                                {
                                    ...mockAttrSimple,
                                    __typename: 'Attribute',
                                    label: {
                                        fr: 'Attr 1'
                                    },
                                    id: 'test_tree_attr'
                                },
                                {
                                    ...mockAttrSimple,
                                    __typename: 'Attribute',
                                    label: {
                                        fr: 'Attr 2'
                                    },
                                    id: 'other_test_tree_attr'
                                }
                            ]
                        }
                    }
                }
            }
        ];
        const onSubmit = jest.fn();
        const onClose = jest.fn();
        const selection = [];

        let comp;
        await act(async () => {
            comp = mount(
                <MockedProvider mocks={mocks}>
                    <AttributesSelectionModal onSubmit={onSubmit} onClose={onClose} openModal selection={selection} />
                </MockedProvider>
            );
            await wait(0);
            comp.update();
        });

        expect(comp).toMatchSnapshot();
    });
});
