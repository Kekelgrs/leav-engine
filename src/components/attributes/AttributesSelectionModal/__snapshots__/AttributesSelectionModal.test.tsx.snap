// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesSelection Snapshot test 1`] = `ReactWrapper {}`;
